src/sema/ast.rs

```diff
@@ -319,6 +321,12 @@ impl Function {
             Some(pt::Mutability::Constant(loc)) => Mutability::View(loc),
         };
 
+        let mangled_name = signature
+            .trim()
```

Comment:
`trim()` should be unnecessary, there should be no whitespace in the signature (it would be a bug if there was)